common/lib/xmodule/xmodule/x_module.py

```diff
@@ -278,6 +278,10 @@ class XModuleMixin(XModuleFields, XBlock):
     # (like a practice problem).
     has_score = False
 
+    # Whether this module can be displayed in read-only mode.  It is safe to set this to True if
+    # all user state is handled through the FieldData API.
+    show_in_read_only_mode = False
```

Comment:
@cpennington This property is supposed to indicate whether the block stores all its state in XBlock fields, or whether it keeps any state outside of that API, which seems to be a property of a block.  Maybe it just needs a better name?

The original plan was to introduce a new view called `read_only_student_view()`, and only render blocks that implement this view.  The default implementation for blocks that only store state in XBlock fields would be to set `read_only_student_view = student_view`.

This turned out to be more involved in practice.  For one, all blocks with children would need additional machinery to call the right view on the child nodes, but there were additional problems, so I decided to use this much simpler approach.